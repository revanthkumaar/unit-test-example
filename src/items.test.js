import React from "react";
import renderer from "react-test-renderer";

import Items from "./Items";

//SNAPSHOT TESTING

test('renders the correct list structure', () => {
    const tree = renderer.create(<Items/>).toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
})

test("renders correctly when there is a single item", () => {
  const items = ["one"];
  const tree = renderer.create(<Items items={items} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly when there are multiple items", () => {
  const items = ["one", "two", "three"];
  const tree = renderer.create(<Items items={items} />).toJSON();
  expect(tree).toMatchSnapshot();
});