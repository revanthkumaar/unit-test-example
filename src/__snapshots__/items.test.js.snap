// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are multiple items 1`] = `
<ul>
  <li
    className="item-message"
  >
    one
  </li>
  <li
    className="item-message"
  >
    two
  </li>
  <li
    className="item-message"
  >
    three
  </li>
</ul>
`;

exports[`renders correctly when there is a single item 1`] = `
<span>
  one
</span>
`;

exports[`renders the correct list structure 1`] = `
<span
  className="data"
>
  No items in list
</span>
`;
